File: array_and_hashing/two_sum.py
def two_sum(nums: list[int], target: int) -> list[int]:
    hashmap = {}
    for i in range(len(nums)):
        complement = target - nums[i]
        if complement in hashmap:
            return [i, hashmap[complement]]
        hashmap[nums[i]] = i
    return []

File: array_and_hashing/two_sum.ts
function twoSum(nums: number[], target: number): number[] {
    const hashmap = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (hashmap.has(complement)) {
            return [i, hashmap.get(complement)!];
        }
        hashmap.set(nums[i], i);
    }

    return [];
}
